packages/block-library/src/pattern/index.php

```diff
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Server-side rendering of the `core/pattern` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ *  Registers the `core/pattern` block on the server.
+ *
+ * @return void
+ */
+function register_block_core_pattern() {
+	register_block_type_from_metadata(
+		__DIR__ . '/pattern',
+		array(
+			'render_callback' => 'render_block_core_pattern',
+		)
+	);
+}
+
+/**
+ * Renders the `core/pattern` block on the server.
+ *
+ * @param array $attributes Block attributes.
+ *
+ * @return string Returns the output of the pattern.
+ */
+function render_block_core_pattern( $attributes ) {
+	$slug = $attributes['slug'];
+	if ( class_exists( 'WP_Block_Patterns_Registry' ) && WP_Block_Patterns_Registry::get_instance()->is_registered( $slug ) ) {
+		$pattern = WP_Block_Patterns_Registry::get_instance()->get_registered( $slug );
+		return do_blocks( '<div>' . $pattern['content'] . '</div>' );
```

Comment:
To avoid styling inconsistencies when converting to a Group, perhaps we should add the Group block's CSS class here:

```suggestion
		return do_blocks( '<div class="wp-block-group">' . $pattern['content'] . '</div>' );
```

Reply:
Technically it is not a Group block until it's converted. Not sure how semantic this would be.

Reply:
Correct, it's not a Group until it's converted + edited, so I don't think we should add the group block CSS class here.

Reply:
Could we make it render an actual Group block, though? In other words, would it be possible to generate and render an actual Group block on the front-end?

Reply:
Yes it's possible, an earlier iteration of the PR did [that](https://github.com/WordPress/gutenberg/pull/33217/commits/0cb327b1635baaffe1c0334f4c42c81e6611f93f), but the feedback was given that a plain div is better / more semantic, since the Pattern block is not the same thing as a Group block. 

This is an example of the default state (no edits made to the Pattern block's contents): 

Site Editor | View
------ | ------
<img width="1392" alt="Screen Shot 2021-10-28 at 3 33 25 PM" src="https://user-images.githubusercontent.com/5375500/139323339-d387a925-02bf-464e-8acd-b6ec0a49fed0.png"> |<img width="1392" alt="Screen Shot 2021-10-28 at 3 35 08 PM" src="https://user-images.githubusercontent.com/5375500/139323474-f69b51f5-1a85-455d-b3de-c6469516e73a.png">

I do think it is a bit tricky to wrap the head around, but why would it make more sense to output a group on the front-end if the contents haven't been edited?

Reply:
I've noticed some themes tend to target Group blocks with styles like additional padding/margin, which they don't apply to plain `<div>`s. In fact, I've even written Custom CSS like that within the past week or so.

If the Pattern block doesn't always output a Group, then editing it could cause a bunch of spacing to appear/disappear out of nowhere on some themes, which would be quite confusing for users.

Reply:
And it's not just basic margin/padding, either. Some themes have special handling for Wide/Full block alignments when inside a Group versus not inside a Group, so this could cause some really bizarre layout shifts in those situations.

Reply:
> I've noticed some themes tend to target Group blocks with styles like additional padding/margin, which they don't apply to plain <div>s. In fact, I've even written Custom CSS like that within the past week or so

These seem like reasons for not rendering the Pattern inside a Group. I think the most likely scenario is for a user to not edit this block, since currently its main use cases for are string translation or including a dynamic URL inside templates. If it were wrapped inside a group by default, the risk of side effects from themes with CSS targeting Groups seems higher. 

Open to other's reviewers perspectives here!

Reply:
Also, this is all new (FSE themes) so we shouldn't base our decisions around practices in existing themes more about best practices.

I'm not sure styling raw groups is a good idea as Group block is meant to be just a div in its essence.

That said, I don't feel strongly here and I don't think it's that important either for a start. Feedback later will help us make the best decision.